git-branchless/src/commands/move.rs

```diff
@@ -187,33 +383,13 @@ pub fn r#move(
             let dest_children: CommitSet = dag
                 .query()
                 .children(CommitSet::from(dest_oid))?
-                .difference(&dag.obsolete_commits);
+                .difference(&dag.obsolete_commits)
+                // In case a source OID is already a child of dest_oid.
+                .difference(&source_oids)
+                .difference(&exact_oids);
```

Comment:
```suggestion
                // In case a source OID is already a child of dest_oid.
                .difference(&source_oids)
                .difference(&exact_oids)
                .difference(&dag.obsolete_commits);
```

Same regarding performance and ordering.